api/auth.py

```diff
@@ -28,11 +28,17 @@ async def authenticate(form: LoginRequest) -> ApiResponse[TokenResponse]:
     """Perform login for a user"""
     return ApiResponse.create(_perform_login(form))
     
-@route.post('/auth/creds')
+@route.put('/auth/creds')
 async def credentials(me: Annotated[User, Depends(AuthManager.get_user_dependency(oauth2_scheme))],
```

Comment:
This should be idempotent since it will always overwrite the credentials for a given user and provider, per RESTful spec that means a PUT